tests/sampling/test_mcmc_external.py

```diff
@@ -86,3 +86,45 @@ def test_step_args():
         )
 
     npt.assert_almost_equal(idata.sample_stats.acceptance_rate.mean(), 0.5, decimal=1)
+
+
+@pytest.mark.parametrize("nuts_sampler", ["pymc", "nutpie", "blackjax", "numpyro"])
+def test_sample_var_names(nuts_sampler):
+    kwargs = {
+        "nuts_sampler": nuts_sampler,
+        "chains": 1,
+    }
+
+    # Generate data
+    seed = 1234
+    rng = np.random.default_rng(seed)
+
+    group = rng.choice(list("ABCD"), size=100)
+    x = rng.normal(size=100)
+    y = rng.normal(size=100)
+
+    group_values, group_idx = np.unique(group, return_inverse=True)
+
+    coords = {"group": group_values}
+
+    # Create model
+    with pm.Model(coords=coords) as model:
+        b_group = pm.Normal("b_group", dims="group")
+        b_x = pm.Normal("b_x")
+        mu = pm.Deterministic("mu", b_group[group_idx] + b_x * x)
+        sigma = pm.HalfNormal("sigma")
+        pm.Normal("y", mu=mu, sigma=sigma, observed=y)
+
+    # Sample with and without var_names, but always with the same seed
+    with model:
+        idata_1 = pm.sample(tune=100, draws=100, random_seed=seed, **kwargs)
```

Comment:
also do `progress_bar=False` and `compute_convergence_checks=False` to reduce test outputs
```suggestion
        idata_1 = pm.sample(tune=100, draws=100, random_seed=seed, **kwargs)
```

Reply:
Added to `kwargs`